index.js

```diff
@@ -196,14 +204,18 @@ function colorDelta(img1, img2, k, m, yOnly) {
         b2 = blend(b2, a2);
     }
 
-    const y = rgb2y(r1, g1, b1) - rgb2y(r2, g2, b2);
+    const y1 = rgb2y(r1, g1, b1);
+    const y2 = rgb2y(r2, g2, b2);
+    const y = y1 - y2;
 
     if (yOnly) return y; // brightness difference only
 
     const i = rgb2i(r1, g1, b1) - rgb2i(r2, g2, b2);
     const q = rgb2q(r1, g1, b1) - rgb2q(r2, g2, b2);
 
-    return 0.5053 * y * y + 0.299 * i * i + 0.1957 * q * q;
+  	const delta = 0.5053 * y * y + 0.299 * i * i + 0.1957 * q * q;
+
+    return y1 > y2 ? -Math.abs(delta) : delta;
```

Comment:
@mourner Not sure if this is what you had in mind. Is there a more elegant way to do this without having the caller convert the value back from the negative?

Reply:
At this point `delta` can only be non-negative so you can drop `Math.abs` here.